core/src/main/java/net/sourceforge/jnlp/util/whitelist/WhitelistEntry.java

```diff
@@ -0,0 +1,99 @@
+package net.sourceforge.jnlp.util.whitelist;
+
+import net.adoptopenjdk.icedteaweb.Assert;
+
+import java.net.URL;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * ...
+ */
+public class WhitelistEntry {
+    private static final String WILDCARD = "*";
+    private static final String HOST_PART_REGEX = "\\.";
+
+    private final String rawWhitelistEntry;
+    private final URL effectiveWhitelistEntry;
```

Comment:
does it make sense to have a `URL` type here? Even if it works now (and I do not get how `*` could be a valid port) it might break in future when we need to change something...

Reply:
No, but changing this would be a major refactor.
I love to do this but would prefer to do it in a separate PR

Reply:
ok, than let's stay with URL for now